core/templates/domain/collection/guest-collection-progress.service.ts

```diff
@@ -65,10 +63,15 @@ export class GuestCollectionProgressService {
   }
 
   getValidCompletedExplorationIds(collection: Collection): string[] {
-    var collectionId = collection.getId();
-    var guestCollectionProgress = this.loadGuestCollectionProgress();
-    var completedExplorationIds = (
-      guestCollectionProgress.getCompletedExplorationIds(collectionId));
+    const collectionId = collection.getId();
+    const guestCollectionProgress = this.loadGuestCollectionProgress();
+    let completedExplorationIds: string[];
+    if (collectionId) {
+      completedExplorationIds = (
+        guestCollectionProgress.getCompletedExplorationIds(collectionId));
+    } else {
+      throw new Error('Collection does not exist!');
+    }
```

Comment:
Let's get rid of some branching 
```suggestion
    if (collectionId == null) {
      throw new Error('Collection does not exist!');
    }
    completedExplorationIds = (
      guestCollectionProgress.getCompletedExplorationIds(collectionId));
```

Reply:
done